src/mame/machine/p2000t_mdcr.h

```diff
@@ -0,0 +1,192 @@
+// license:BSD-3-Clause
+// copyright-holders:Erwin Jansen
+/**********************************************************************
+
+    Philips P2000t  Mini Digital Cassette Recorder Emulation
+
+**********************************************************************
+
+                    +12V      1      8       !WCD
+            OV (signal)       2      9       !REV
+            OV (power)        3      A       !FWD
+                    GND       4      B       RDC
+                   !WDA       6      C       !RDA
+                   !BET       7      D       !CIP
+                                     E       !WEN
+
+**********************************************************************/
+
+#ifndef MAME_MACHINE_P2000T_MDCR_H
+#define MAME_MACHINE_P2000T_MDCR_H
+
+#pragma once
+
+#include "imagedev/cassette.h"
+#include <queue>
+
+/// \brief A Phase Decoder used in a Philips MDCR220 Mini Cassette Recorder
+///
+/// A phase decoder is capable of converting a signal stream into a
+/// a series of bits that go together with a clock signal. This phase
+/// decoder is conform to what you would find in an Philips MDCR220
+///
+/// Signals are converted into bits whenever the line signal
+/// changes from low to high and vice versa on a clock signal.
+///
+/// A transition on a clock boundary from low to high is a 1.
+/// A transition on a clock boundary from high to low is a 0
+/// An intermediate transition halfway between the clock boundary
+/// can occur when there are consecutive 0s or 1s. See the example
+/// below where the clock is marked by a |
+///
+///          1    0    1    1    0    0
+///   RDA:  _|----|____|--__|----|__--|__--
+///   RDC:  _|-___|-___|-___|-___|-___|-___
+///          ^                      ^
+///          |-- clock signal       |-- intermediate transition.
+///
+/// This particular phase decoder expects a signal of
+/// 1010 1010 which is used to derive the clock T.
+/// after a reset.
+class phase_decoder
```

Comment:
This class name is too generic for the global namespace – please move this class inside the `mdcr_device` class or give it a less generic name.

Reply:
will fix